imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,197 @@
+        assert packet.data["SC_HIT_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_CODICE_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_LO_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_HI_45_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_HI_90_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_ULTRA_45_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_ULTRA_90_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_SWE_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_IDEX_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_GLOWS_STATUS"].derived_value == "NOT_OPERATIONAL"
+        assert packet.data["SC_SPINPERIODVALID"].derived_value == "INVALID"
+        assert packet.data["SC_SPINPHASEVALID"].derived_value == "INVALID"
+        assert packet.data["SC_ATTITUDE"].derived_value == "SUNSENSOR"
+        assert packet.data["SC_CATBEDHEATERFLAG"].derived_value == "ON"
+        assert packet.data["SC_AUTONOMY"].derived_value == "OPERATIONAL"
+        assert packet.data["HIT_STATUS"].derived_value == "OFF-NOMINAL"
+        assert packet.data["SWE_NOM_FLAG"].derived_value == "OFF-NOMINAL"
+        assert packet.data["SWE_OPS_FLAG"].derived_value == "NON-HVSCI"
+
+
+def test_generate_xarray(binary_packet_path, xtce_ialirt_path, decom_packets_data):
+    """This function checks that all instrument parameters are accounted for."""
+
+    instrument_lengths = {}
+    instruments = ["SC", "HIT", "MAG", "COD_LO", "COD_HI", "SWE", "SWAPI"]
+
+    xarray_data = generate_xarray(binary_packet_path, xtce_ialirt_path)
+
+    # Assert that all parameters are the same
+    # between packet data and generated xarray
+    for instrument in instruments:
+        instrument_list = list(xarray_data[instrument].coords)
+        instrument_list.extend(list(xarray_data[instrument].data_vars))
+
+        # Create a dictionary of the number of parameters for each instrument
+        instrument_lengths[instrument] = len(instrument_list)
+
+        packet_data_keys_set = set(decom_packets_data[0].data.keys())
+        instrument_list_set = set(instrument_list)
+
+        # Assert that the instrument parameters from the xarray are
+        # a subset of the packet data
+        assert instrument_list_set.issubset(packet_data_keys_set) is True
+
+    # Assert that the length of parameters is the same
+    # between packet data and generated xarray
+    assert sum(instrument_lengths.values()) == len(
+        list(decom_packets_data[0].data.keys())
+    )
+
+    # Check that all the dimensions are the correct length
+    # example:len(xarray_data['HIT'].coords['HIT_SC_TICK'].values) == 32
+    for instrument in instruments:
+        dimension_name = next(iter(xarray_data[instrument].dims))
```

Comment:
Do you need to get all dimension names? Right now I think you're just getting the first one.
```suggestion
        for dimension_name in xarray_data[instrument].dims:
```